analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -93,6 +93,30 @@ public void Execute()
         checks.ExecutionCompleted();
     }
 
+    private bool CheckVisitCount(ExplodedNode node, int visitCount)
+    {
+        return visitCount <= MaxOperationVisits
+            || (visitCount <= MaxOperationVisits + 1 && IsLoopCondition());
+
+        bool IsLoopCondition() =>
+            node.Block.BranchValue is not null  // avoid further checks for blocks without BranchValue
```

Comment:
Comment not needed anymore
```suggestion
            node.Block.BranchValue is not null
```